packages/adblocker/src/preprocessor.ts

```diff
@@ -0,0 +1,248 @@
+import { StaticDataView, sizeOfUTF8 } from './data-view';
+
+export type EnvKeys =
+  | 'ext_ghostery'
+  | 'ext_abp'
+  | 'ext_ublock'
+  | 'ext_ubol'
+  | 'ext_devbuild'
+  | 'env_chromium'
+  | 'env_edge'
+  | 'env_firefox'
+  | 'env_mobile'
+  | 'env_safari'
+  | 'env_mv3'
+  | 'false'
+  | 'cap_html_filtering'
+  | 'cap_user_stylesheet'
+  | 'adguard'
+  | 'adguard_app_android'
+  | 'adguard_app_ios'
+  | 'adguard_app_mac'
+  | 'adguard_app_windows'
+  | 'adguard_ext_android_cb'
+  | 'adguard_ext_chromium'
+  | 'adguard_ext_edge'
+  | 'adguard_ext_firefox'
+  | 'adguard_ext_opera'
+  | 'adguard_ext_safari'
+  | (string & {});
+
+export class Env extends Map<EnvKeys, boolean> {
+  constructor() {
+    super();
+
+    // Control the default conditions via constructor
+    this.set('true', true);
+    this.set('false', false);
+  }
+}
+
+export const enum PreprocessorTypes {
```

Comment:
Nit. These are not really preprocessors but tokens involved in the preprocessor construct?

Reply:
Renamed to `PreprocessorTokens`.